File: ios/Types/PlacePickerResult.swift
//
//  PlacePickerResult.swift
//  react-native-place-picker
//
//  Created by b0iq on 11/11/2022.
//

import ExpoModulesCore

struct PlacePickerResult: Record {
    @Field
    var coordinate: PlacePickerCoordinate
    @Field
    var address: PlacePickerAddress?
    @Field
    var didCancel: Bool
    @Field
    var radius: Double?
    @Field
    var radiusCoordinates: RadiusCoordinates?
}

struct BoundsCoordinates: Record {
    @Field
    var northeast: PlacePickerCoordinate
    @Field
    var southwest: PlacePickerCoordinate
}

struct RadiusCoordinates: Record {
    @Field
    var center: PlacePickerCoordinate
    @Field
    var bounds: BoundsCoordinates
}

File: android/src/main/java/expo/modules/placepicker/Types/PlacePickerResult.kt
package expo.modules.placepicker

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class PlacePickerResult : Record {
    @Field
    var coordinate: PlacePickerCoordinate? = null

    @Field
    var address: PlacePickerAddress? = null

    @Field
    var didCancel: Boolean? = null

    @Field
    var radius: Double? = null

    @Field
    var radiusCoordinates: RadiusCoordinates? = null
}

class BoundsCoordinates : Record {
    @Field
    var northeast: PlacePickerCoordinate? = null
    @Field
    var southwest: PlacePickerCoordinate? = null
}

class RadiusCoordinates : Record {
    @Field
    var center: PlacePickerCoordinate? = null
    @Field
    var bounds: BoundsCoordinates? = null
}
